godot-core/src/builtin/vector3.rs

```diff
@@ -336,3 +340,55 @@ pub enum Vector3Axis {
 impl GodotFfi for Vector3Axis {
     ffi_methods! { type sys::GDExtensionTypePtr = *mut Self; .. }
 }
+
+impl Index<usize> for Vector3 {
+    type Output = f32;
+
+    fn index(&self, index: usize) -> &Self::Output {
+        match index {
+            0 => &self.x,
+            1 => &self.y,
+            2 => &self.z,
+            _ => panic!("Index {index} out of bounds"),
+        }
+    }
+}
+
+impl IndexMut<usize> for Vector3 {
+    fn index_mut(&mut self, index: usize) -> &mut Self::Output {
+        match index {
+            0 => &mut self.x,
+            1 => &mut self.y,
+            2 => &mut self.z,
+            _ => panic!("Index {index} is out of bounds"),
+        }
+    }
+}
+
+impl GlamType for Vec3 {
+    type Mapped = Vector3;
+
+    fn to_front(&self) -> Self::Mapped {
+        Vector3::new(self.x, self.y, self.z)
+    }
+
+    fn from_front(mapped: &Self::Mapped) -> Self {
+        Vec3::new(mapped.x, mapped.y, mapped.z)
+    }
+}
+
+impl GlamType for Vec3A {
+    type Mapped = Vector3;
+
+    fn to_front(&self) -> Self::Mapped {
+        Vector3::new(self.x, self.y, self.z)
+    }
+
+    fn from_front(mapped: &Self::Mapped) -> Self {
+        Vec3A::new(mapped.x, mapped.y, mapped.z)
+    }
+}
```

Comment:
We are not using `Vec3A` at the moment, so this is dead code.

Reply:
we will be because the transform3d will use `Affine3A`, since there is no `Affine3`, which uses `Vec3A` internally

Reply:
Ok, we then need to see if `Vec3A` <-> `Vector3` conversions are not necessary all over the place 🤔 otherwise we might not gain much from using `glam`. But I guess it's a good start.